language_reference_manual.md

```diff
@@ -1,15 +1,73 @@
-# Language Reference Manual: Sift
+# LANGUAGE REFERENCE MANUAL: SIFT
 
-## Lexical Conventions
+## LEXICAL CONVENTIONS
 
-## Types
+## TYPES
 
-## Operators 
+## OPERATORS 
 
-## Statements & Expressions
+## STATEMENTS & EXPRESSIONS
 
-## Garbage Collection 
+## MEMORY MANAGEMENT 
 
-## NLP Features 
+As a text processing language, memory management becomes a critical aspect. But unlike C, Sift has automatic memory management.
+With our automatic memory management, it's faster to develop programs without being bothered about the low-level memory details. 
+It also prevents the program from memory leaks and dangling pointers.
 
-## Grammar
\ No newline at end of file
+Sift uses generational garbage collection for automatic memory management. Sift supports three generation of in total. An object
+moves into an older generation whenever it survives a garbage collection process on its current generation.
+
+The garbage collector keeps track of all objects in the memory. When a new object is created, it starts its life in the first generation.
+There is a predefined value for the theshold number of objects for each generation. If it exceeds that threshold, the garbage collector triggers
+a collection process. For objects that survive this process, they are moved into the older generation.
+
+### Using The Garbage Collector Module
+
+Programmers can change this default behavior of Garbage collection depending upon their available resources and requirement of their program.
```

Comment:
I would also add something above like 

`Garbage collection is available to Sift users in the form of an importable module gc with a specific set of functionality.`